### PR TITLE
Reintroduce git info to jpsreport

### PR DESCRIPTION
**Pull request by Ozaq**
jpsreport now again properly reports git commit and change when running.
This was a left-over from the integration.

Fixes: #572

### CODE REVIEW

**Comment by gjaeger** [comment]
It seems to be working. 

```zsh
$ jpsreport --version

----
JuPedSim - JPSreport

Current date   : Sat Dec 14 11:20:17 2019
Version        : 0.8.5
Compiler       : clang++ (11.0.0)
Commit hash    : v0.8.4-881-gb6522ee5
Commit date    : Sat Dec 14 10:56:33 2019
Branch         : pr604
Python         : /usr/local/bin/python3 (3.7.5)
----
```

File: jpsreport/general/ArgumentParser.cpp [hunk 1]
```diff
@@ -102,11 +102,11 @@ void Logs()
...
-    // Log->Write("Version        : %s", JPSREPORT_VERSION);
+    Log->Write("Version        : %s", JPSREPORT_VERSION);
     Log->Write("Compiler       : %s (%s)", true_cxx.c_str(), true_cxx_ver.c_str());
-    // Log->Write("Commit hash    : %s", GIT_COMMIT_HASH);
-    // Log->Write("Commit date    : %s", GIT_COMMIT_DATE);
-    // Log->Write("Branch         : %s", GIT_BRANCH);
+    Log->Write("Commit hash    : %s", GIT_COMMIT_HASH);
+    Log->Write("Commit date    : %s", GIT_COMMIT_DATE);
+    Log->Write("Branch         : %s", GIT_BRANCH);
     Log->Write("Python         : %s (%s)\n----\n", PYTHON, PYTHON_VERSION);
```
Comment (gjaeger) [inline jpsreport/general/ArgumentParser.cpp]: This output is no longer required. We removed the option to plot timeseries and Voronoi diagrams with python.

File: jpsreport/general/ArgumentParser.cpp [hunk 2]
```diff
@@ -102,11 +102,11 @@ void Logs()
     // first logs will go to stdout
     Log->Write("----\nJuPedSim - JPSreport\n");
     Log->Write("Current date   : %s", currentTime.c_str());
-    // Log->Write("Version        : %s", JPSREPORT_VERSION);
+    Log->Write("Version        : %s", JPSREPORT_VERSION);
     Log->Write("Compiler       : %s (%s)", true_cxx.c_str(), true_cxx_ver.c_str());
-    // Log->Write("Commit hash    : %s", GIT_COMMIT_HASH);
-    // Log->Write("Commit date    : %s", GIT_COMMIT_DATE);
-    // Log->Write("Branch         : %s", GIT_BRANCH);
+    Log->Write("Commit hash    : %s", GIT_COMMIT_HASH);
```
Comment (gjaeger) [inline jpsreport/general/ArgumentParser.cpp]: Is there a _tag_ mistake?
`Commit hash    : v0.8.4-881-gb6522ee5`

File: jpsreport/CMakeLists.txt [hunk 3]
```diff
@@ -319,11 +319,13 @@ target_link_libraries(report PUBLIC
     tinyxml
     fs
     Boost::boost
+    git-info
 )
 
 target_compile_definitions(report PUBLIC
     PYTHON="${PYTHON_EXECUTABLE}"
     PYTHON_VERSION="${PYTHON_VERSION_STRING}"
+    JPSREPORT_VERSION="0.8.5"
```
Comment (gjaeger) [inline jpsreport/CMakeLists.txt]: Why do you set the version here manually?

File: CMakeLists.txt [hunk 4]
```diff
@@ -224,7 +224,8 @@ else()
     set(GIT_BRANCH "UNKNONW")
 endif()
 
-list(APPEND GIT_BUILD_INFO_DEFINES
+add_library(git-info INTERFACE)
+target_compile_definitions(git-info INTERFACE
```
Comment (gjaeger) [inline CMakeLists.txt]: Do jpscore and jpsreport use these lines together?

File: jpsreport/general/ArgumentParser.cpp (hunk 1 above)
Comment (Ozaq) [reply jpsreport/general/ArgumentParser.cpp]: I can remove it but it does not really have anything to do with this PR since I didn't change the line

File: jpsreport/CMakeLists.txt (hunk 3 above)
Comment (Ozaq) [reply jpsreport/CMakeLists.txt]: Ah right, I wanted to ask you or @zeroset if both report and core should use the same version number but then I forgot. So should they share the version number?

File: jpsreport/general/ArgumentParser.cpp (hunk 2 above)
Comment (Ozaq) [reply jpsreport/general/ArgumentParser.cpp]: It is the same pattern as used in jps-report
Read it as:
{last tag}-{commits since last tag}-{short commit id}

File: CMakeLists.txt (hunk 4 above)
Comment (Ozaq) [reply CMakeLists.txt]: So it works like this:
in line 227 a new "interface" library is defined that allows to propagate compilation flags, include paths, preprocessor definitions etc. by linking to this library
So jpsreport and jpscore depend on this library (see the respective target_link_libraries calls) and the defines will be set during the compilation of the source files.
TL;DR; yes :)

File: jpsreport/general/ArgumentParser.cpp (hunk 1 above)
Comment (gjaeger) [reply jpsreport/general/ArgumentParser.cpp]: Okay. I remember it for a future pull request.

File: jpsreport/general/ArgumentParser.cpp (hunk 2 above)
Comment (gjaeger) [reply jpsreport/general/ArgumentParser.cpp]: ok. Thanks.

File: jpsreport/CMakeLists.txt (hunk 3 above)
Comment (gjaeger) [reply jpsreport/CMakeLists.txt]: A good question for @zeroset and @chraibi.

File: CMakeLists.txt (hunk 4 above)
Comment (gjaeger) [reply CMakeLists.txt]: Thanks.

File: jpsreport/CMakeLists.txt (hunk 3 above)
Comment (Ozaq) [reply jpsreport/CMakeLists.txt]: I talked to @zeroset in chat and the conclusion was that core and report will use the same verison number from now on

Comment (gjaeger) [reply jpsreport/CMakeLists.txt]: ok

File: jpsreport/general/ArgumentParser.cpp (hunk 1 above)
Comment (Ozaq) [reply jpsreport/general/ArgumentParser.cpp]: @gjaeger I will fix this when cleaning up the cmakelists. See #574
